homeassistant/components/griddy/config_flow.py

```diff
@@ -0,0 +1,87 @@
+"""Config flow for Griddy Power integration."""
+import logging
+
+from aiohttp import ClientError
+from griddypower.async_api import LOAD_ZONES, AsyncGriddy
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.helpers import aiohttp_client
+
+from .const import CONF_LOADZONE
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema({vol.Required(CONF_LOADZONE): vol.In(LOAD_ZONES)})
+
+
+class GriddyGateway:
+    """A griddy gateway object."""
+
+    def __init__(self, clientsession):
+        """Gateway with a clientsession."""
+        self._clientsession = clientsession
+
+    async def async_verify_connection(self, settlement_point):
+        """Connect to griddy and make a basic request."""
+        data = await AsyncGriddy(
+            self._clientsession, settlement_point=settlement_point
+        ).async_getnow()
+        return bool(data.now.datetime)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    griddy_gateway = GriddyGateway(aiohttp_client.async_get_clientsession(hass))
+
+    try:
+        await griddy_gateway.async_verify_connection(data[CONF_LOADZONE])
+    except ClientError:
+        raise CannotConnect
+
+    # Return info that you want to store in the config entry.
+    return {"title": f"Load Zone {data[CONF_LOADZONE]}"}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Griddy Power."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        info = None
+        if user_input is not None:
+            try:
+                info = await validate_input(self.hass, user_input)
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+
+            if "base" not in errors:
+                await self.async_set_unique_id(user_input[CONF_LOADZONE])
+                self._abort_if_unique_id_configured()
+                return self.async_create_entry(title=info["title"], data=user_input)
+
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_import(self, user_input):
+        """Handle import."""
+        await self.async_set_unique_id(user_input[CONF_LOADZONE])
```

Comment:
If we set unique id here and then pass to the user step, which also sets unique id and uses the abort if unique id configured helper, the flow will abort, right?

Reply:
Testing:

Setup LZ_HOUSTON in UI

Added in yaml

```
griddy:
  loadzone: LZ_HOUSTON
```

Restarted. No errors and no config flows.  grepped the log for anything griddy or \.py.  Did two test cycles and restarts for good measure since I changed the unique_id to _s

Reply:
Yeah, sorry, I mixed things up.